lgc/builder/llpcYCbCrAddressHandler.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+***********************************************************************************************************************
+* @file  llpcYCbCrAddressHandler.cpp
+* @brief LLPC source file: Implementation of LLPC class YCbCrAddressHandler
+***********************************************************************************************************************
+*/
+#include "llpcYCbCrAddressHandler.h"
+#include "llpcGfxRegHandler.h"
+#include "llpcInternal.h"
+
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+using namespace lgc;
+using namespace llvm;
+
+// =====================================================================================================================
+// Generate base address of each image plane
+void YCbCrAddressHandler::GenBaseAddress(
+    uint32_t planeNum) // The count of planes
```

Comment:
nit: maybe `// The number of planes` to stay consistent with the argument name?

Reply:
How many planes can we have? What happens when I call this with  `planeNum > 3`? Should some plane numbers be disallowed?

Reply:
> nit: maybe `// The number of planes` to stay consistent with the argument name?

Okay, this name is fine.

> How many planes can we have? What happens when I call this with `planeNum > 3`? Should some plane numbers be disallowed?

At least 1 plane and at most 3 planes, I've added some assert, though the result should not care planeNum > 3 when generating these plane base address.